### PR TITLE
Remove trim_start_matches targeting '.'

### PR DESCRIPTION
**Pull request by jrray**
Use a custom method for trimming '/' or "./" that doesn't change ".foo"
into "foo".

Fixes https://github.com/spkenv/spk/issues/1036

### CODE REVIEW

**Comment by jrray** [comment]
I thought this was a problem internal to `Manifest` and made the new private method there, but then found that `spfs-vfs` has the same bug.

File: crates/spfs/src/tracking/diff.rs [hunk 1]
```diff
@@ -117,60 +117,70 @@ pub fn compute_diff<U1: Clone>(a: &Manifest<U1>, b: &Manifest<U1>) -> Vec<Diff<U
...
     changes
 }
 
+// Allow: most instances will be of the large variant; boxing is
+// counter-productive.
+#[allow(clippy::large_enum_variant)]
+enum DiffPathResult<U1, U2> {
+    Diff(Diff<U1, U2>),
+    PathMissingFromBothManifests,
+    UselessMask,
```
Comment (rydrman) [inline crates/spfs/src/tracking/diff.rs:143]: I'm not entirely clear on this variant based on the naming, would you mind adding some documentation about when this result appears?

Comment (jrray) [reply crates/spfs/src/tracking/diff.rs:143]: Did you see "path was masked in the right manifest but didn't exist in the left" above?

That's what I'd add as the description.

Comment (rydrman) [reply crates/spfs/src/tracking/diff.rs:143]: Oh, yes I see that does help as context

**Review by rydrman — APPROVED** [review]

**Pull request merged**